features/unmarshal/unmarshal.go

```diff
@@ -122,32 +127,49 @@ func (p *unmarshal) GenerateHelpers() {
 			`}`)
 	})
 
-	p.P(`
-		var (
-			ErrInvalidLength = `, p.Ident("fmt", "Errorf"), `("proto: negative length found during unmarshaling")
-			ErrIntOverflow = `, p.Ident("fmt", "Errorf"), `("proto: integer overflow")
-			ErrUnexpectedEndOfGroup = `, p.Ident("fmt", "Errorf"), `("proto: unexpected end of group")
-		)
-	`)
+	p.Helper("err", func(p *generator.GeneratedFile) {
+		p.P(`
+			var (
+				ErrInvalidLength = `, p.Ident("fmt", "Errorf"), `("proto: negative length found during unmarshaling")
+				ErrIntOverflow = `, p.Ident("fmt", "Errorf"), `("proto: integer overflow")
+				ErrUnexpectedEndOfGroup = `, p.Ident("fmt", "Errorf"), `("proto: unexpected end of group")
+			)`)
+	})
+
+	if p.unsafe {
+		p.Helper("unsafeBytesToString", func(p *generator.GeneratedFile) {
+			p.P(`
+				func unsafeBytesToString(b []byte) string {
+					return *(*string)(`, p.Ident("unsafe", `Pointer`), `(&b))
+				}`)
+		})
```

Comment:
I think we should bump `vtprotobuf` to require at least Go 1.20 and use [`unsafe.String`](https://pkg.go.dev/unsafe#String) here. I don't think the method used here is going to break anytime soon, or probably ever, but it seems more correct to me to use the standard library for this functionality which is inherently unsafe.

Reply:
Thanks for taking a look already! I think it makes sense, I've updated the version (in https://github.com/planetscale/vtprotobuf/pull/103/commits/b366c923fd89ac37963f78eda1e01194cc941eaf) and switched to using `unsafe.String` in https://github.com/planetscale/vtprotobuf/pull/103/commits/6a7e8af003d6096480a9ec07b15f729426f8de8c. Since `unsafe.SliceData` and `unsafe.StringData` are also available, I've updated the test as well because these functions are preferred to `reflect.StringHeader` and `reflect.SliceHeader` (they are getting deprecated).